src/applicationManager.ts

```diff
@@ -15,7 +18,22 @@ export type ApplicationAndServicePrincipalId = {
   applicationId: string;
   servicePrincipalObjectId: string;
 };
-
+export type AppRoleAndGroupAssignments = {
+  displayName: string;
+  description: string;
+  value: string;
+  id: string;
+  groups: Array<string>;
+};
+export type AppRole = {
+  allowedMemberTypes: Array<string>;
```

Comment:
more normal (although functionally equivalent)
```suggestion
  allowedMemberTypes: string[];
```